docs/core/diagnostics/microsoft-diagnostics-netcore-client.md

```diff
@@ -0,0 +1,459 @@
+# Microsoft.Diagnostics.NETCore.Client
+
+**This article applies to: ✔️** .NET Core 3.0 SDK and later versions
+
+Microsoft.Diagnostics.NETCore.Client (also known as the Diagnostics Client library) is a managed library that lets you interact with .NET Core runtime (CoreCLR) for various diagnostics related tasks, such as tracing via [EventPipe](eventpipe.md), requesting a dump, or attaching an ICorProfiler. This library is the backing library behind many diagnostics tools such as [`dotnet-counters`](dotnet-counters.md), [`dotnet-trace`](dotnet-trace.md), [`dotnet-gcdump`](dotnet-gcdump.md), and [`dotnet-dump`](dotnet-dump.md). Using this library, you can write your own diagnostics tools customized for your particular scenario.
+
+## Install
+
+You can acquire [Microsoft.Diagnostics.NETCore.Client](https://www.nuget.org/packages/Microsoft.Diagnostics.NETCore.Client/) by adding a `PackageReference` to your project. The package is hosted on `NuGet.org`. 
+
+## Use Microsoft.Diagnostics.NETCore.Client to write your own diagnostic tools
+
+These samples show how to use Microsoft.Diagnostics.NETCore.Client library. Some of these examples also show parsing the event payloads by using [TraceEvent](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.TraceEvent/) library.
+
+### Attach to a process and print out all the runtime GC events in real time to the console
+
+This snippet shows how to start an EventPipe session using the [.NET runtime provider](../../fundamentals/diagnostics/runtime-events.md) with the GC keyword at informational level and how to use the `EventPipeEventSource` class provided by the [TraceEvent library](https://www.nuget.org/packages/Microsoft.Diagnostics.Tracing.TraceEvent/) to parse the incoming events and print their names to the console in real time.
+
+```cs
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using System.Collections.Generic;
+
+public void PrintRuntimeGCEvents(int processId)
+{
```

Comment:
All of the code in this article is pretty big, it should be moved and referenced as a snippet. See the paragraph that starts with **or code snippets, create a subfolder called snippets inside the folder that contains your article** on https://docs.microsoft.com/en-us/contribute/dotnet/dotnet-contribute#make-a-contribution-to-net-docs

You can also look at this article as an example:

- Article: https://github.com/dotnet/docs/blob/master/docs/core/tutorials/library-with-visual-studio.md
- Snippets folder for article: https://github.com/dotnet/docs/tree/master/docs/core/tutorials/snippets/library-with-visual-studio